File: backend-Nijan/src/main/java/com/ecommerce/backendNijan/service/OrderService.java
package com.ecommerce.backendNijan.service;

import com.ecommerce.backendNijan.entity.CartsEntity;
import com.ecommerce.backendNijan.model.ICart;
import com.ecommerce.backendNijan.model.OrderDto;
import com.ecommerce.backendNijan.response.PageResponse;

import java.util.List;

public interface OrderService {

    void insert(OrderDto orderDto);

    PageResponse<OrderDto> getAllByOrder(int pageNo, int pageSize, String sortBy, String sortDirection,
                                           String searchValue);

    OrderDto getAllByOrderId(Long orderId);

    List<OrderDto> getAllByUserId(Long userId);

    boolean update(Long cartId, String cartStatus);

    List<ICart> getAll();
}

File: vite-nijan/src/component/modules/OrderService.js
import httpCommon from "./http-common";

class OrderService {

    insert(order) {
        return httpCommon.post("http://localhost:8888/api/order/insert", order);
    }

    getAll(page) {
        return httpCommon.get('http://localhost:8888/api/order/get-all' + "?page_no=" + page.pageNo + "&page_size=" + page.pageSize
        + "&sort_direction=" + page.sortDirection + "&sort_by=" + page.sortBy);
    }

    update(cartId, orderStatus) {
        return httpCommon.put("http://localhost:8888/api/order/update/" + cartId + "/" + orderStatus);
    }
}

export default new OrderService();